src/test/lib/operators/pqp_utils_test.cpp

```diff
@@ -66,17 +72,72 @@ TEST_F(PQPUtilsTest, VisitPQPNonConstOperators) {
 }
 
 TEST_F(PQPUtilsTest, VisitPQPConstOperators) {
-  auto node_b = std::make_shared<const GetTable>("bar");
-  auto node_c = std::make_shared<const Limit>(node_b, to_expression(int64_t{1}));
-  const auto expected_nodes = std::vector<std::shared_ptr<const AbstractOperator>>{node_c, node_b};
+  const auto node_b = std::make_shared<const GetTable>("bar");
+  const auto node_c = std::make_shared<const Limit>(node_b, value_(1));
 
+  const auto expected_nodes = std::vector<std::shared_ptr<const AbstractOperator>>{node_c, node_b};
   auto actual_nodes = std::vector<std::shared_ptr<const AbstractOperator>>{};
-  visit_pqp(expected_nodes[0], [&](const auto& node) {
+
+  visit_pqp(node_c, [&](const auto& node) {
     actual_nodes.emplace_back(node);
     return PQPVisitation::VisitInputs;
   });
 
   EXPECT_EQ(actual_nodes, expected_nodes);
 }
 
+TEST_F(PQPUtilsTest, ResolveUncorrelatedSubquery) {
+  const auto table_wrapper = std::make_shared<TableWrapper>(Projection::dummy_table());
+  table_wrapper->execute();
+
+  EXPECT_EQ(resolve_uncorrelated_subquery(table_wrapper), AllTypeVariant(0));
+}
+
+TEST_F(PQPUtilsTest, ResolveUncorrelatedSubqueryWrongOperatorState) {
+  const auto table_wrapper = std::make_shared<TableWrapper>(Projection::dummy_table());
+
+  // Operator is not executed yet.
+  EXPECT_THROW(resolve_uncorrelated_subquery(table_wrapper), std::logic_error);
+
+  table_wrapper->execute();
+  table_wrapper->clear_output();
+
+  // Operator is cleared.
+  EXPECT_THROW(resolve_uncorrelated_subquery(table_wrapper), std::logic_error);
+}
+
+TEST_F(PQPUtilsTest, ResolveUncorrelatedSubqueryEmptyResult) {
+  const auto dummy_table = Table::create_dummy_table({{"a", DataType::Int, false}});
+  const auto table_wrapper = std::make_shared<TableWrapper>(dummy_table);
+  table_wrapper->execute();
+
+  EXPECT_TRUE(variant_is_null(resolve_uncorrelated_subquery(table_wrapper)));
```

Comment:
I'd guess the result is an empty table here. Why is the variant null?

Reply:
Even more confusing when the only column is not nullable.

Reply:
Even more confusing when the only column is not nullable.

Reply:
Even more confusing when the only column is not nullable.

Reply:
If the table is empty, it means we do not get a subquery result. Since we use this function in a TableScan, this (slightly hacky) behavior creates a ValueScan with a NULL value as predicate value, which does not yield a result. This mimics the case of the Join to Predicate rewrite with an empty table on one side:

```SQL
SELECT n_name
  FROM nation, region
 WHERE n_regionkey = r_regionkey
   AND r_name = 'NotAnActualRegion'
```
--> empty Join result

```SQL
SELECT n_name
  FROM nation
 WHERE n_regionkey = (SELECT r_regionkey
                        FROM region
                       WHERE r_name = 'NotAnActualRegion')
```
--> empty subquery, leads to scan `n_regionkey = NULL`, which is also empty

Reply:
Can you add a short comment?

Reply:
Is this sufficient?

https://github.com/hyrise/hyrise/blob/3652455dc6c428434c319d417871af35940b4abd/src/test/lib/operators/pqp_utils_test.cpp#L109-L112

Reply:
Jep, thanks.